app/src/main/java/com/example/tasky/feature_login/presentation/LoginScreen.kt

```diff
@@ -28,52 +27,50 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import androidx.navigation.NavController
 import androidx.navigation.compose.rememberNavController
-import com.example.tasky.CalendarNav
+import com.example.tasky.AgendaNav
 import com.example.tasky.R
-import com.example.tasky.common.domain.isValidEmail
-import com.example.tasky.common.domain.isValidPassword
 import com.example.tasky.common.presentation.CreateErrorAlertDialog
 import com.example.tasky.common.presentation.Header
 import com.example.tasky.common.presentation.SimpleButton
 import com.example.tasky.common.presentation.TextBox
 import com.example.tasky.feature_login.domain.model.AuthenticationViewState
-import com.example.tasky.feature_login.domain.model.LoginUserInfo
 
 
 @Composable
-fun LoginScreenContent(navController: NavController) {
-    // todo figure out why text fields are not saving w/orientation change
-    val loginViewModel: LoginViewModel = hiltViewModel()
-    val authenticationViewModel: AuthenticationViewModel = hiltViewModel()
+fun LoginScreenContent(
+    navController: NavController,
+    loginViewModel: LoginViewModel = hiltViewModel()
+) {
+    val uiState by loginViewModel.uiState.collectAsStateWithLifecycle()
+    val viewState by loginViewModel.viewState.collectAsStateWithLifecycle()
 
-    val viewState by authenticationViewModel.viewState.collectAsState()
     val email by loginViewModel.email.collectAsState()
     val password by loginViewModel.password.collectAsState()
 
     val isEmailValid by loginViewModel.isEmailValid.collectAsState()
     val isPasswordValid by loginViewModel.isPasswordValid.collectAsState()
+    val isPasswordVisible by loginViewModel.isPasswordVisible.collectAsState()
```

Comment:
I'd recommend to use `collectAsStateWithLifecycle()` that since it's the safest way to collect flows in Android. For that, you'll need this dependency:
```
implementation "androidx.lifecycle:lifecycle-runtime-compose:2.x.x"
```
Then use it like this:
```
val state by viewModel.state.collectAsStateWithLifecycle()
```
If you have a simple flow it shouldn't make a difference, but as soon as you convert a normal `Flow` into a `StateFlow` with `stateIn`, the Flow operators will be executed even when the app is in the background (which you typically don't want when dealing with UI updates).